src/binary-search-tree.js

```diff
@@ -12,27 +13,49 @@ class BinarySearchTree {
   // assigns it to either the left or right subtree,
   // depending on its value
   insert(value) {
-
+    const branch = (value > this.value) ? 'right' : 'left';
+    if (this[branch] !== null) {
+      this[branch].insert(value);
+    } else {
+      this[branch] = new BinarySearchTree(value);
+    }
   }
   // Checks the binary search tree for the input target
   // Can be written recursively or iteratively
   contains(target) {
-
+    if (this.value === target) return true;
+    if (target > this.value && this.right !== null) return this.right.contains(target);
+    if (this.left !== null) return this.left.contains(target);
+    return false;
   }
   // Traverses the tree in a depth-first manner, i.e. from top to bottom
   // Applies the given callback to each tree node in the process
   depthFirstForEach(cb) {
-
+    cb(this.value);
+    if (this.left !== null) this.left.depthFirstForEach(cb);
```

Comment:
Here you can just check `if (this.left)` and `if (this.right)`, which will check that `this.left` and `this.right` are not `null` or `undefined`, among other things.